library/src/webapp/editor/ckextraplugins/common-wl/js/folder-listing.js

```diff
@@ -28,9 +29,20 @@ $.fn.folderListing = function(options) {
 
   var urlPrefix = '/direct/content/resources/';
 
-  var getHtmlResults = function(json, $div) {
+  var getHtmlResults = function(json, $div, root) {
     var html = $('<ul/>').addClass('jqueryFileTree').hide();
-
+    var urlRoot = json[0].resourceId.trim();
+    if(settings.displayRootDirectory && urlRoot === root ){
```

Comment:
Actually, probably better to have a `container` variable, and by default set it to be equal to `html` but if we are showing the root folder set container equal to `ul`. That way you keep the switching all in the same place.